### DISCUSSION TITLE
how to find subarray?

### DISCUSSION THREAD

**Discussion by WingDust** [ramda/ramda#3386] [Q&A] [answered]
Given the array
 
`y =[0,0,0,10,1,10,76,178,1020,0,10,0,229,2,2,0,0,178,1020,10,0,178,0,2,127,76,1020,178,178,1020,0,1020,0,76,1020,130,0,0]`

 to find all subarrays that start and end with 0 and must contain non-zero numbers in between. 

Return the start and end indices of these subarrays.

**Comment by bunglegrind** [comment, marked as answer]
quick and dirty

```
const isStart = ([f, s]) => f === 0 && s !== 0;
const isEnd = ([f, s]) => f !== 0 && s === 0;
const reduceIndexed = R.addIndex(R.reduce);

reduceIndexed((acc, item, index)  =>  (
    isStart(item)
    ? [...acc, index]
    : (
       isEnd(item)
       ?  [...acc, index + 1]
       : acc
      )
), [], R.aperture(2, y));
```
[REPL](https://ramdajs.com/repl/#?const%20y%20%3D%5B0%2C0%2C0%2C10%2C1%2C10%2C76%2C178%2C1020%2C0%2C10%2C0%2C229%2C2%2C2%2C0%2C0%2C178%2C1020%2C10%2C0%2C178%2C0%2C2%2C127%2C76%2C1020%2C178%2C178%2C1020%2C0%2C1020%2C0%2C76%2C1020%2C130%2C0%2C0%5D%3B%0A%0A%2F%2FR.pipe%28%0A%2F%2F%20%20R.splitWhenever%28R.equals%280%29%29%2C%0A%2F%2F%20%20R.map%28%28v%29%20%3D%3E%20%5B0%2C%20...v%2C%200%5D%29%20%2F%2For%20R.map%28R.pipe%28R.prepend%280%29%2C%20R.append%280%29%29%29%0A%2F%2F%29%28y%29%3B%0A%0Aconst%20isStart%20%3D%20%28%5Bf%2C%20s%5D%29%20%3D%3E%20f%20%3D%3D%3D%200%20%26%26%20s%20%21%3D%3D%200%3B%0Aconst%20isEnd%20%3D%20%28%5Bf%2C%20s%5D%29%20%3D%3E%20f%20%21%3D%3D%200%20%26%26%20s%20%3D%3D%3D%200%3B%0Aconst%20reduceIndexed%20%3D%20R.addIndex%28R.reduce%29%3B%0A%0AreduceIndexed%28%28acc%2C%20item%2C%20index%29%20%20%3D%3E%20%20%28%0A%20%20%20%20isStart%28item%29%0A%20%20%20%20%3F%20%5B...acc%2C%20index%5D%0A%20%20%20%20%3A%20%28%0A%20%20%20%20%20%20%20isEnd%28item%29%0A%20%20%20%20%20%20%20%3F%20%20%5B...acc%2C%20index%20%2B%201%5D%0A%20%20%20%20%20%20%20%3A%20acc%0A%20%20%20%20%20%20%29%0A%29%2C%20%5B%5D%2C%20R.aperture%282%2C%20y%29%29%3B%0A)

I suppose a better solution is achievable by employing transducers
